src/map/clif.c

```diff
@@ -15683,33 +15692,55 @@ void clif_quest_send_mission(struct map_session_data *sd)
 
 /// Notification about a new quest (ZC_ADD_QUEST).
 /// 02b3 <quest id>.L <active>.B <start time>.L <expire time>.L <mobs>.W { <mob id>.L <mob count>.W <mob name>.24B }*3
+/// 09f9 <quest id>.L <active>.B <start time>.L <expire time>.L <mobs>.W { <hunt identification>.L <mob type>.L <mob id>.L <min level>.L <max level>.L <mob count>.W <mob name>.24B }*3
 void clif_quest_add(struct map_session_data *sd, struct quest *qd)
 {
-	int fd;
-	int i;
+	int i, len, real_len;
+	uint8 *buf = NULL;
+	struct packet_quest_add_header *packet = NULL;
 	struct quest_db *qi;
 
 	nullpo_retv(sd);
 	nullpo_retv(qd);
-	fd = sd->fd;
+	
+	len = sizeof(struct packet_quest_add_header)
+	            + MAX_QUEST_OBJECTIVES * sizeof(struct packet_quest_hunt_sub); // >= than the actual length
+
+	packet = (struct packet_quest_add_header *)WBUFP(buf, 0);
```

Comment:
This is a NULL pointer (`buf` is NULL)

Reply:
+ 	buf = aMalloc(len);
I tried to fix the BUF, but the instruction "SETQUEST" still prevented NEXT from appearing.

Reply:
@idamonli what you mean by Prevented next from appearing?